pkg/network/pod/annotations/generator_test.go

```diff
@@ -520,6 +520,22 @@ var _ = Describe("Annotations Generator", func() {
 			clusterConfig = newClusterConfig(kv)
 		})
 
+		It("Should not generate network attachment annotation when there are no networks", func() {
+			vmi := libvmi.New(
+				libvmi.WithNamespace(testNamespace),
+				libvmi.WithAutoAttachPodInterface(false),
```

Comment:
How is this relevant? I think this has influence only when the VM controller computes the VMI spec.
It was also not there before and I think the implementation its is not effected by this knob.

Reply:
This configuration option is required to get to this scenario in a running system and was added for completeness.
It does not affect this logic, so if you think it adds noise I can remove it.

Reply:
On a running system, that is correct for a VM but not for a VMI. I’m in favor of completeness but in this case I find it confusing.

Reply:
> On a running system, that is correct for a VM but not for a VMI.

The VMI mutator uses it when creating a VMI on a live system.

Reply:
https://github.com/kubevirt/kubevirt/blob/a724aff0139167ee3997d55ee90cbbb4439aafcd/pkg/virt-api/webhooks/mutating-webhook/mutators/vmi-mutator.go#L85

https://github.com/kubevirt/kubevirt/blob/a724aff0139167ee3997d55ee90cbbb4439aafcd/pkg/defaults/defaults.go#L111

https://github.com/kubevirt/kubevirt/blob/a724aff0139167ee3997d55ee90cbbb4439aafcd/pkg/defaults/defaults.go#L232

Reply:
Thank you for correcting me. I am surprised again and again about how nasty reality is.

I would like to drop this behavior from the VMI.